### DISCUSSION TITLE
useIfiniteQuery with additional query params

### DISCUSSION THREAD

**Discussion by renanfcamara** [TanStack/query#4647] [Q&A]
Hi, im currently developing an timeline page with useIfiniteQuery and an InfiniteScroll

i have 3 possible params to my endpoint: page, sort and search

i can add sort and search to my dependency key, but when i scroll down to fetch more data, my endpoint is fetching with olders params but page.

Here is my code:

```javascript

const [ sort, setSort ] = useState({ timelineDate: -1 });
const [ search, setSearch ] = useState('');
const [ page, setPage ] = useState(1);

const handleChangeSort = async (newSort) => {
  setSort(newSort);
}

const handleChangeSearch = (newSearch) => {
  setSearch(newSearch);
}

const { data, fetchNextPage, hasNextPage, isFetching, isLoading } = useInfiniteQuery({
    queryKey: getTimelineKey(search, sort), 
    queryFn: ({ pageParam = 1 }) => timelineService.getTimeline(pageParam, sort, search),
    getNextPageParam: (lastPage, allPages) => lastPage.paging.hasNextPage ? lastPage.paging.page + 1 : undefined
})

<button onClick={() => handleChangeSort({ favorites: -1 }) }>Filter to favorites</button>


```

```javacript
<InfiniteScrollExample
  next={fetchNextPage}
  hasMore={hasNextPage}
  scrollThreshold={'90%'}
>
```

![Screen Shot 2022-12-15 at 12 38 49 PM](https://user-images.githubusercontent.com/89042043/207903258-072922cc-1a5b-4ab0-96ee-f3f575d9110a.png)
After change my sort to `{ favorites: -1 } `

![Screen Shot 2022-12-15 at 12 40 28 PM](https://user-images.githubusercontent.com/89042043/207903594-57ce50b9-a86e-463f-b007-83c74e188bda.png)
payload when fires pagination

Im wondering, there's any way to preserve my `sort` and `search`?

**Comment by TkDodo** [comment]
code looks good. show a codesandbox reproduction please
